tests/unit-tests/margo-addr.c

```diff
@@ -192,26 +208,29 @@ static MunitResult test_margo_addr_to_string(const MunitParameter params[], void
     struct test_context* ctx = (struct test_context*)data;
 
     hret = margo_addr_self(ctx->mid, &addr);
-    munit_assert_int(hret, ==, HG_SUCCESS);
-    munit_assert_not_null(addr);
+    munit_assert_int_goto(hret, ==, HG_SUCCESS, error);
+    munit_assert_not_null_goto(addr, error);
 
     char addr_str[256];
     hg_size_t addr_str_size = 256;
     memset(addr_str, 0, 256);
 
     hret = margo_addr_to_string(ctx->mid, addr_str, &addr_str_size, addr);
-    munit_assert_int(hret, ==, HG_SUCCESS);
-    munit_assert_string_not_equal(addr_str, "\0");
+    munit_assert_int_goto(hret, ==, HG_SUCCESS, error);
+    munit_assert_string_not_equal_goto(addr_str, "\0", error);
 
     hret = margo_addr_free(ctx->mid, addr);
-    munit_assert_int(hret, ==, HG_SUCCESS);
+    munit_assert_int_goto(hret, ==, HG_SUCCESS, error);
 
     addr_str_size = 256;
     memset(addr_str, 0, 256);
     hret = margo_addr_to_string(ctx->mid, addr_str, &addr_str_size, HG_ADDR_NULL);
-    munit_assert_int(hret, ==, HG_INVALID_ARG);
+    munit_assert_int_goto(hret, ==, HG_INVALID_ARG, error);
 
     return MUNIT_OK;
+
+error:
```

Comment:
Actually, all of these test cases just do a failure return at the label.  Is there a reason the macros don't just return error?  Or is it just to allow for the possibility to clean up state in future cases?

Reply:
I hesitated in just doing a return, but in the more general cases, you might want to do some cleanup before actually returning.

Reply:
Fair enough.  It should produce a warning if we were to try to use the macro without putting a label with that name in the function, right?

Reply:
It's not going to compile if the label isn't present.